sklearn/utils/tests/test_stats.py

```diff
@@ -0,0 +1,66 @@
+import numpy as np
+from numpy.testing import assert_array_equal
+
+from sklearn.utils.stats import _weighted_percentile
+
+
+def test_weighted_percentile():
+    y = np.empty(102, dtype=np.float64)
+    y[:50] = 0
+    y[-51:] = 2
+    y[-1] = 100000
+    y[50] = 1
+    sw = np.ones(102, dtype=np.float64)
+    sw[-1] = 0.0
+    score = _weighted_percentile(y, sw, 50)
+    assert score == 1
+
+    # Check for multioutput
```

Comment:
`_weighted_percentile` is not necessarily about "output" so this comment is a bit obscure.

I'd rather a test along the lines of:
```
def test_weighted_percentile_2d():
   X = ...
   sw = ...
   for each value of _weighted_percentile(X, sw):
       assert that _weighted_percentile(X[:, i], sw) matches it
```